### PR TITLE
Select Support

### PR DESCRIPTION
**Pull request by LordOfPolls**
## About this pull request

Adds select support, and requisite documentation 

## Changes

- Add `selected_options` to ctx
- To prevent future issues, with new stuff, exposed ctx['data'] in context
- Added docs to helper functions
- Added in limits to selects in `discord_slash.utils`
- Added example to `components.rst`

## Checklist

- [x] I've run the `pre_push.py` script to format and lint code.
- [x] I've checked this pull request runs on `Python 3.6.X`.
- [ ] This fixes something in [Issues](https://github.com/eunwoo1104/discord-py-slash-command/issues).
    - Issue:
- [x] This adds something new.
- [ ] There is/are breaking change(s).
- [x] (If required) Relevant documentation has been updated/added.
- [ ] This is not a code change. (README, docs, etc.)

### CODE REVIEW

**Comment by LordOfPolls** [comment]
While im waiting for the reviews, id like to take the time to say 

ｓｃｒｅｗ　ｔｉｍｅｚｏｎｅｓ

**Review by benwoo1110 — COMMENTED** [review]
Looks fine to me (eye power), tho have not tested the changes. Also, have some minor queries on docs as detailed below.

File: discord_slash/context.py [hunk 1]
```diff
@@ -23,13 +23,16 @@ class InteractionContext:
     :ivar bot: discord.py client.
     :ivar _http: :class:`.http.SlashCommandRequest` of the client.
     :ivar _logger: Logger instance.
+    :ivar data: The raw data of the interaction.
+    :ivar values: The values sent with the interaction. Currently for selects.
```
[the remark this answers is not in the document]
Comment (LordOfPolls) [reply discord_slash/context.py:27]: The interaction is `sent` so imo `sent` is more accurate

File: discord_slash/utils/manage_components.py [hunk 2]
```diff
@@ -156,17 +156,21 @@ def create_select_option(
     """
     Creates an option for select components.
 
-    .. warning::
-        Currently, select components are not available for public use, nor do they have official documentation. The parameters will not be documented at this time.
-
-    :param label: The label of the option.
-    :param value: The value that the bot will recieve when this option is selected.
+    :param label: The user-facing name of the option.
```
[the remark this answers is not in the document]
Comment (LordOfPolls) [reply discord_slash/utils/manage_components.py]: Works for me, you want to make a suggestion to add it, or shall i commit

File: discord_slash/context.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (benwoo1110) [reply discord_slash/context.py:27]: ah ok

File: discord_slash/utils/manage_components.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (benwoo1110) [reply discord_slash/utils/manage_components.py]: you can commit 👍

**Review by benwoo1110 — APPROVED** [review]
Basic function of select working for me:
image

code I used to test for reference https://gist.github.com/benwoo1110/76ebfa5bfa8b363fddde9ef1a4f69c76

**Comment by artem30801** [comment]
LGTM

**Pull request merged**